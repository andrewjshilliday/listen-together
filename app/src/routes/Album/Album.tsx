import React, { useState, useEffect, createRef, useLayoutEffect, useRef } from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import { MusicKitService, MusicKitApiService } from '../../services';
import { Loading, TrackList, MediaItemCardCarousel } from '../../components/common';
import styled from 'styled-components';

const Album: React.FC = (props: any) => {
  const id = props.match.params.id;
  const [loading, setLoading] = useState(true);
  const [album, setAlbum] = useState({} as MusicKit.MediaItem);
  const [artistAlbums, setArtistAlbums] = useState([] as MusicKit.MediaItem[]);
  const [duration, setDuration] = useState('');
  const albumRef = useRef<MusicKit.MediaItem>(album);
  const notesRef = createRef<HTMLDivElement>();

  useEffect(() => {
    const getAlbum = async () => {
      setLoading(true);
      setAlbum(await MusicKitApiService.Album(id));
      setLoading(false);
    };
    getAlbum();
  }, [id]);

  useEffect(() => {
    if (album.relationships) {
      let durationInMillis
      let hours = 0, minutes = 0;
      let durationString = '';
      
      durationInMillis = moment.duration(album.relationships.tracks.data.map((i: MusicKit.MediaItem) => i.attributes.durationInMillis)
        .reduce((a: number, b: number) => a + b, 0));
      hours = Math.floor(durationInMillis.asHours());
      durationInMillis = moment.duration(durationInMillis.asMilliseconds() - (hours * 60 * 60 * 1000));
      minutes = Math.round(durationInMillis.asMinutes());

      if (hours) {
        durationString += `${hours} Hour`;
        if (hours > 1) { durationString += 's' }
        durationString += ' '
      }
      durationString += `${minutes} Minute`;
      if (minutes > 1) { durationString += 's' }

      setDuration(durationString);
    }

    if (album.relationships?.artists) {
      MusicKitApiService.Artist(album.relationships.artists.data[0].id, 'albums').then(res => {
        setArtistAlbums(res.relationships.albums.data);
      })
    }

    albumRef.current = album;
  }, [album])

  useLayoutEffect(() => {
    setEditorialNotesStyle();
    window.addEventListener('resize', setEditorialNotesStyle);
    return () => window.removeEventListener('resize', setEditorialNotesStyle);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [notesRef]);
  
  const setEditorialNotesStyle = () => {
    if (!albumRef.current?.attributes?.editorialNotes || !notesRef.current) {
      return;
    }

    const height = window.innerHeight;
    const notesOffsetTop = notesRef.current.getBoundingClientRect().top;
    notesRef.current.style.maxHeight = `${height - notesOffsetTop - 110}px`;
  };

  if (loading) return (<Loading />);
  
  return (
    <AlbumContainer>
      <AlbumTracksContainer>
        <SidebarContainer>
          <StickySidebar>
            <img src={MusicKitService.FormatArtwork(album.attributes.artwork, 500)} className="img-fluid rounded" alt={album.attributes.name} />
            <p>{album.relationships.tracks.data.length} Songs, {duration}</p>
            {
              album.attributes.editorialNotes &&
              <>
                <hr />
                <EditorialNotes ref={notesRef}>
                {
                  album.attributes.editorialNotes.standard ?
                  <span dangerouslySetInnerHTML={{ __html: `${album.attributes.editorialNotes.standard}` }}></span> :
                  <span dangerouslySetInnerHTML={{ __html: `${album.attributes.editorialNotes.short}`}}></span>
                }
                </EditorialNotes>
              </>
            }
          </StickySidebar>
        </SidebarContainer>
        <TrackListContainer>
          <TrackListHeader>
            <h1 className="text-truncate">{album.attributes.name}</h1>
            <ArtistGenre>
              <Link to={`/artist/${album.relationships.artists.data[0].id}`}><h2 className="text-truncate">{album.attributes.artistName}</h2></Link>
              <h2>{album.attributes.genreNames && (<>&nbsp;| {album.attributes.genreNames[0]}</>)}</h2>
            </ArtistGenre>
          </TrackListHeader>
          <TrackList tracks={album.relationships.tracks.data}></TrackList>
        </TrackListContainer>
      </AlbumTracksContainer>
      {
        artistAlbums.length && 
        <ArtistAlbumsContainer>
          <MediaItemCardCarousel items={artistAlbums} title={`More By ${album.attributes.artistName}`}></MediaItemCardCarousel> 
        </ArtistAlbumsContainer>
      }
    </AlbumContainer>
  );
}

export default Album;


const AlbumContainer = styled.div``;
const AlbumTracksContainer = styled.div`
  display: flex;
`;
const SidebarContainer = styled.div`
  width: 30%;
  min-width: 250px;
  max-width: 400px;
  padding: 10px 15px;
`;
const StickySidebar = styled.div`
  position: sticky;
  top: 10px;
`;
const EditorialNotes = styled.div`
  overflow: auto;
  font-size: 10pt;
`;
const TrackListContainer = styled.div`
  width: 70%;
  padding-left: 15px;
`;
const TrackListHeader = styled.div`
  position: sticky;
  top: 0;
  padding-bottom: 10px;
  background: white;
`;
const ArtistGenre = styled.div`
  display: flex;
`;
const ArtistAlbumsContainer = styled.div`
  margin: 0 10px;
`;
