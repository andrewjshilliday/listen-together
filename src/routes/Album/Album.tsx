import React, { useState, useEffect, createRef } from 'react';
import { MKAlbum } from '../../services/MusicKitApi';
import { FormatArtwork } from '../../services/MusicKit';
import { TrackList } from '../../components/common';
import CircularProgress from '@material-ui/core/CircularProgress';
import './Album.scss';

const Album: React.FC = (props: any) => {
  const [album, setAlbum] = useState({} as any);
  const [loading, setLoading] = useState(true);
  const id = props.match.params.id;
  const notesRef = createRef<HTMLDivElement>();

  useEffect(() => {
    const getAlbum = async () => {
      setLoading(true);
      setAlbum(await MKAlbum(id));
      setLoading(false);
    };
    getAlbum();
  }, [id]);

  useEffect(() => {
    setEditorialNotesStyle();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [notesRef]);

  window.onresize = () => {
    setEditorialNotesStyle();
  };
  
  const setEditorialNotesStyle = () => {
    if (!album.attributes || !album.attributes.editorialNotes) {
      return;
    }

    if (notesRef.current) {
      const height = window.innerHeight;
      const notesOffsetTop = notesRef.current.getBoundingClientRect().top;
      notesRef.current.style.maxHeight = `${height - notesOffsetTop - 110}px`;
    }
  };

  if (loading) return (<div className="loading"><CircularProgress /></div>);
  
  return (
    <div className="album view">
      <div className="sidebar sticky">
        <img src={FormatArtwork(album.attributes.artwork, 500)} className="img-fluid rounded" alt={album.attributes.name} />
        {
          album.attributes.editorialNotes &&
          <>
            <hr />
            <div ref={notesRef} className="editorial-notes">
            {
              album.attributes.editorialNotes.standard ?
              <span dangerouslySetInnerHTML={{ __html: `${album.attributes.editorialNotes.standard}` }}></span> :
              <span dangerouslySetInnerHTML={{ __html: `${album.attributes.editorialNotes.short}`}}></span>
            }
            </div>
          </>
        }
      </div>
      <div className="track-list-container">
        <div className="track-list-header">
          <h1 className="text-truncate">{album.attributes.name}</h1>
          <h2 className="text-truncate">{album.attributes.artistName} {album.attributes.genreNames && (<>| {album.attributes.genreNames[0]}</>)}</h2>
        </div>
        <TrackList tracks={album.relationships.tracks.data} displayArtwork={false}></TrackList>
      </div>
    </div>
  );
}

export default Album;
