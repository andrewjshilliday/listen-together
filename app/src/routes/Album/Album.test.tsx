import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import renderer, { act } from 'react-test-renderer';
import Album from './Album';

jest.mock('../../services/MusicKit');
jest.mock('../../services/MusicKitApi', () => ({
  MusicKitApiService: require('../../__mocks__/services/MusicKitApi/MusicKitApi.mock').default
}));
jest.mock('../../services/ListenTogetherApi', () => ({
  ListenTogetherApiService: require('../../__mocks__/services/ListenTogetherApi/ListenTogetherApi.mock').default
}));

jest.useFakeTimers();

it('renders correctly', async () => {  
  const tree = renderer.create(
    <MemoryRouter initialEntries={[ '/album/1447538299' ]}>
      <Route path="/album/:id" component={Album} />
    </MemoryRouter>);

  jest.runTimersToTime(1);

  await act(async () => {
    await Promise.resolve();
  });

  expect(tree.toJSON()).toMatchSnapshot();
});
