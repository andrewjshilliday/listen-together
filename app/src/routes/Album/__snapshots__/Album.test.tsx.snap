// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-fzoiQi jkvegY"
>
  <div
    className="sc-fzqARJ lnyOvm"
  >
    <div
      className="sc-fzqNqU bhkQpa"
    >
      <div
        className="sc-fzoyTs iVKUqR"
      >
        <img
          alt="This Land"
          className="img-fluid rounded"
        />
        <p>
          17
           Songs, 
          1 Hour 13 Minutes
        </p>
        <hr />
        <div
          className="sc-fzoNJl jdzzfG"
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "We’re calling it: In the rock ’n’ roll history books, Gary Clark Jr. will have two eras: before <i>This Land</i> and after it.<br />
Just get a load of the fire and fury that opens the title track: “F**k you, I’m America’s son/This is where I come from,” he snarls. Clark’s rage is partially directed at his racist neighbour in Austin, Texas, who can’t seem to accept Clark’s sprawling 50-acre ranch, as well as a few experiences from his childhood. “I had a few situations down there with some racism, and some Confederate flags, and people calling me out of their trucks, all that kind of stuff,” he told Beats 1 host Zane Lowe. “I had a beat that I laid down but didn't have any lyrics over it and it just came to me. I just went in there and fired off.” But it's also, more broadly, aimed at President Trump for fanning the flames of racism across the American South. He’s pissed off, and finally speaking out.<br />
<i>This Land</i>, which Clark produced himself, confronts these realities head-on, including stressful community divisions (“What About Us”), touring fatigue (“The Guitar Man”) and political activism (“Feed the Babies”). In an effort to find some common ground, he reminds us why we came to his music in the first place: its soulful, spontaneous spirit. The rallying <i>wooo</i>s and rip-roaring guitars on the standout “Gotta Get Into Something” recall Stiff Little Fingers as much as they do Chuck Berry. And like any rousing punk anthem, it’s its own form of protest song: a thunderous, gritty alarm that dares you to sit still.",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="sc-fzoXWK fniOzM"
    >
      <div
        className="sc-fzpmMD eWMA-di"
      >
        <h1
          className="text-truncate"
        >
          This Land
        </h1>
        <div
          className="sc-fznMAR iKwQaO"
        >
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <h2
              className="text-truncate"
            >
              Gary Clark Jr.
            </h2>
          </a>
          <h2>
             | 
            Rock
          </h2>
        </div>
      </div>
      <div
        className="sc-AxheI fniENP"
      >
        <div
          className="sc-Axmtr cffNzl"
        >
          <div
            className="sc-AxmLO eHiwgP"
            onClick={[Function]}
          >
            <i
              className="fas fa-plus"
            />
          </div>
          <span
            className="sc-fzozJi kXLUYR text-truncate"
          >
            This Land
          </span>
          <span
            className="sc-fzplWN eXRBjQ track-duration"
          >
            5:41
          </span>
        </div>
        <div
          className="sc-Axmtr cffNzl"
        >
          <div
            className="sc-AxmLO eHiwgP"
            onClick={[Function]}
          >
            <i
              className="fas fa-plus"
            />
          </div>
          <span
            className="sc-fzozJi kXLUYR text-truncate"
          >
            What About Us
          </span>
          <span
            className="sc-fzplWN eXRBjQ track-duration"
          >
            4:30
          </span>
        </div>
        <div
          className="sc-Axmtr cffNzl"
        >
          <div
            className="sc-AxmLO eHiwgP"
            onClick={[Function]}
          >
            <i
              className="fas fa-plus"
            />
          </div>
          <span
            className="sc-fzozJi kXLUYR text-truncate"
          >
            I Got My Eyes on You (Locked & Loaded)
          </span>
          <span
            className="sc-fzplWN eXRBjQ track-duration"
          >
            5:11
          </span>
        </div>
        <div
          className="sc-Axmtr cffNzl"
        >
          <div
            className="sc-AxmLO eHiwgP"
            onClick={[Function]}
          >
            <i
              className="fas fa-plus"
            />
          </div>
          <span
            className="sc-fzozJi kXLUYR text-truncate"
          >
            I Walk Alone
          </span>
          <span
            className="sc-fzplWN eXRBjQ track-duration"
          >
            3:44
          </span>
        </div>
        <div
          className="sc-Axmtr cffNzl"
        >
          <div
            className="sc-AxmLO eHiwgP"
            onClick={[Function]}
          >
            <i
              className="fas fa-plus"
            />
          </div>
          <span
            className="sc-fzozJi kXLUYR text-truncate"
          >
            Feelin' Like a Million
          </span>
          <span
            className="sc-fzplWN eXRBjQ track-duration"
          >
            3:34
          </span>
        </div>
        <div
          className="sc-Axmtr cffNzl"
        >
          <div
            className="sc-AxmLO eHiwgP"
            onClick={[Function]}
          >
            <i
              className="fas fa-plus"
            />
          </div>
          <span
            className="sc-fzozJi kXLUYR text-truncate"
          >
            Gotta Get Into Something
          </span>
          <span
            className="sc-fzplWN eXRBjQ track-duration"
          >
            3:04
          </span>
        </div>
        <div
          className="sc-Axmtr cffNzl"
        >
          <div
            className="sc-AxmLO eHiwgP"
            onClick={[Function]}
          >
            <i
              className="fas fa-plus"
            />
          </div>
          <span
            className="sc-fzozJi kXLUYR text-truncate"
          >
            Got to Get Up
          </span>
          <span
            className="sc-fzplWN eXRBjQ track-duration"
          >
            2:37
          </span>
        </div>
        <div
          className="sc-Axmtr cffNzl"
        >
          <div
            className="sc-AxmLO eHiwgP"
            onClick={[Function]}
          >
            <i
              className="fas fa-plus"
            />
          </div>
          <span
            className="sc-fzozJi kXLUYR text-truncate"
          >
            Feed the Babies
          </span>
          <span
            className="sc-fzplWN eXRBjQ track-duration"
          >
            4:46
          </span>
        </div>
        <div
          className="sc-Axmtr cffNzl"
        >
          <div
            className="sc-AxmLO eHiwgP"
            onClick={[Function]}
          >
            <i
              className="fas fa-plus"
            />
          </div>
          <span
            className="sc-fzozJi kXLUYR text-truncate"
          >
            Pearl Cadillac
          </span>
          <span
            className="sc-fzplWN eXRBjQ track-duration"
          >
            5:05
          </span>
        </div>
        <div
          className="sc-Axmtr cffNzl"
        >
          <div
            className="sc-AxmLO eHiwgP"
            onClick={[Function]}
          >
            <i
              className="fas fa-plus"
            />
          </div>
          <span
            className="sc-fzozJi kXLUYR text-truncate"
          >
            When I'm Gone
          </span>
          <span
            className="sc-fzplWN eXRBjQ track-duration"
          >
            3:48
          </span>
        </div>
        <div
          className="sc-Axmtr cffNzl"
        >
          <div
            className="sc-AxmLO eHiwgP"
            onClick={[Function]}
          >
            <i
              className="fas fa-plus"
            />
          </div>
          <span
            className="sc-fzozJi kXLUYR text-truncate"
          >
            The Guitar Man
          </span>
          <span
            className="sc-fzplWN eXRBjQ track-duration"
          >
            4:26
          </span>
        </div>
        <div
          className="sc-Axmtr cffNzl"
        >
          <div
            className="sc-AxmLO eHiwgP"
            onClick={[Function]}
          >
            <i
              className="fas fa-plus"
            />
          </div>
          <span
            className="sc-fzozJi kXLUYR text-truncate"
          >
            Low Down Rolling Stone
          </span>
          <span
            className="sc-fzplWN eXRBjQ track-duration"
          >
            4:18
          </span>
        </div>
        <div
          className="sc-Axmtr cffNzl"
        >
          <div
            className="sc-AxmLO eHiwgP"
            onClick={[Function]}
          >
            <i
              className="fas fa-plus"
            />
          </div>
          <span
            className="sc-fzozJi kXLUYR text-truncate"
          >
            The Governor
          </span>
          <span
            className="sc-fzplWN eXRBjQ track-duration"
          >
            2:21
          </span>
        </div>
        <div
          className="sc-Axmtr cffNzl"
        >
          <div
            className="sc-AxmLO eHiwgP"
            onClick={[Function]}
          >
            <i
              className="fas fa-plus"
            />
          </div>
          <span
            className="sc-fzozJi kXLUYR text-truncate"
          >
            Don't Wait Til Tomorrow
          </span>
          <span
            className="sc-fzplWN eXRBjQ track-duration"
          >
            4:05
          </span>
        </div>
        <div
          className="sc-Axmtr cffNzl"
        >
          <div
            className="sc-AxmLO eHiwgP"
            onClick={[Function]}
          >
            <i
              className="fas fa-plus"
            />
          </div>
          <span
            className="sc-fzozJi kXLUYR text-truncate"
          >
            Dirty Dishes Blues
          </span>
          <span
            className="sc-fzplWN eXRBjQ track-duration"
          >
            5:03
          </span>
        </div>
        <div
          className="sc-Axmtr cffNzl"
        >
          <div
            className="sc-AxmLO eHiwgP"
            onClick={[Function]}
          >
            <i
              className="fas fa-plus"
            />
          </div>
          <span
            className="sc-fzozJi kXLUYR text-truncate"
          >
            Highway 71 (Bonus Track)
          </span>
          <span
            className="sc-fzplWN eXRBjQ track-duration"
          >
            3:31
          </span>
        </div>
        <div
          className="sc-Axmtr cffNzl"
        >
          <div
            className="sc-AxmLO eHiwgP"
            onClick={[Function]}
          >
            <i
              className="fas fa-plus"
            />
          </div>
          <span
            className="sc-fzozJi kXLUYR text-truncate"
          >
            Did Dat (Bonus Track)
          </span>
          <span
            className="sc-fzplWN eXRBjQ track-duration"
          >
            6:42
          </span>
        </div>
      </div>
      <p
        className="sc-fznxKY jWZJuF"
      >
        Released: 
        Friday, February 22, 2019
         
        <br />
        Warner Records
         | 
        ℗ 2019 Warner Records Inc.
      </p>
    </div>
  </div>
  <div
    className="sc-fznWOq eUIqzb"
  >
    <div
      className="sc-AxirZ fJUMyY"
    >
      <h2>
        More By Gary Clark Jr.
      </h2>
      <div
        className="sc-AxiKw enaPwj"
      >
        <div
          className="sc-AxhCb iEIFwb"
          onClick={[Function]}
        />
        <div
          className="sc-AxgMl eFueIV"
        >
          <div
            className="sc-AxjAm dYpDUQ"
          >
            <a
              href="/album/1447538299"
              onClick={[Function]}
            >
              <img
                alt="This Land"
                className="img-fluid rounded"
              />
              <span
                className="text-truncate"
              >
                This Land
              </span>
            </a>
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </div>
          <div
            className="sc-AxjAm dYpDUQ"
          >
            <a
              href="/album/1022109426"
              onClick={[Function]}
            >
              <img
                alt="The Story of Sonny Boy Slim"
                className="img-fluid rounded"
              />
              <span
                className="text-truncate"
              >
                The Story of Sonny Boy Slim
              </span>
            </a>
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </div>
          <div
            className="sc-AxjAm dYpDUQ"
          >
            <a
              href="/album/906127510"
              onClick={[Function]}
            >
              <img
                alt="Gary Clark Jr. Live"
                className="img-fluid rounded"
              />
              <span
                className="text-truncate"
              >
                Gary Clark Jr. Live
              </span>
            </a>
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </div>
          <div
            className="sc-AxjAm dYpDUQ"
          >
            <a
              href="/album/560639480"
              onClick={[Function]}
            >
              <img
                alt="Blak and Blu"
                className="img-fluid rounded"
              />
              <span
                className="text-truncate"
              >
                Blak and Blu
              </span>
            </a>
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </div>
          <div
            className="sc-AxjAm dYpDUQ"
          >
            <a
              href="/album/1200488004"
              onClick={[Function]}
            >
              <img
                alt="Live North America 2016"
                className="img-fluid rounded"
              />
              <span
                className="text-truncate"
              >
                Live North America 2016
              </span>
            </a>
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </div>
          <div
            className="sc-AxjAm dYpDUQ"
          >
            <a
              href="/album/560652745"
              onClick={[Function]}
            >
              <img
                alt="Blak and Blu (Deluxe Version)"
                className="img-fluid rounded"
              />
              <span
                className="text-truncate"
              >
                Blak and Blu (Deluxe Version)
              </span>
            </a>
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </div>
          <div
            className="sc-AxjAm dYpDUQ"
          >
            <a
              href="/album/455141519"
              onClick={[Function]}
            >
              <img
                alt="The Bright Lights - EP"
                className="img-fluid rounded"
              />
              <span
                className="text-truncate"
              >
                The Bright Lights - EP
              </span>
            </a>
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </div>
          <div
            className="sc-AxjAm dYpDUQ"
          >
            <a
              href="/album/626872587"
              onClick={[Function]}
            >
              <img
                alt="iTunes Session"
                className="img-fluid rounded"
              />
              <span
                className="text-truncate"
              >
                iTunes Session
              </span>
            </a>
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </div>
          <div
            className="sc-AxjAm dYpDUQ"
          >
            <a
              href="/album/1494805000"
              onClick={[Function]}
            >
              <img
                alt="This Land (Remix) [From the Tonight Show Starring Jimmy Fallon] [feat. Black Thought] - Single"
                className="img-fluid rounded"
              />
              <span
                className="text-truncate"
              >
                This Land (Remix) [From the Tonight Show Starring Jimmy Fallon] [feat. Black Thought] - Single
              </span>
            </a>
            <span
              className="text-truncate"
            >
              Gary Clark Jr. & The Roots
            </span>
          </div>
          <div
            className="sc-AxjAm dYpDUQ"
          >
            <a
              href="/album/1490025532"
              onClick={[Function]}
            >
              <img
                alt="Pearl Cadillac (feat. Andra Day) - Single"
                className="img-fluid rounded"
              />
              <span
                className="text-truncate"
              >
                Pearl Cadillac (feat. Andra Day) - Single
              </span>
            </a>
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </div>
          <div
            className="sc-AxjAm dYpDUQ"
          >
            <a
              href="/album/1476285802"
              onClick={[Function]}
            >
              <img
                alt="Buddha (feat. Gary Clark Jr.) - Single"
                className="img-fluid rounded"
              />
              <span
                className="text-truncate"
              >
                Buddha (feat. Gary Clark Jr.) - Single
              </span>
            </a>
            <span
              className="text-truncate"
            >
              Macy Gray
            </span>
          </div>
          <div
            className="sc-AxjAm dYpDUQ"
          >
            <a
              href="/album/1458664025"
              onClick={[Function]}
            >
              <img
                alt="Can't Stop the Bleeding (feat. Gary Clark Jr. & Gramatik) - Single"
                className="img-fluid rounded"
              />
              <span
                className="text-truncate"
              >
                Can't Stop the Bleeding (feat. Gary Clark Jr. & Gramatik) - Single
              </span>
            </a>
            <span
              className="text-truncate"
            >
              Tom Morello
            </span>
          </div>
          <div
            className="sc-AxjAm dYpDUQ"
          >
            <a
              href="/album/1447106653"
              onClick={[Function]}
            >
              <img
                alt="This Land - Single"
                className="img-fluid rounded"
              />
              <span
                className="text-truncate"
              >
                This Land - Single
              </span>
            </a>
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </div>
          <div
            className="sc-AxjAm dYpDUQ"
          >
            <a
              href="/album/1420363046"
              onClick={[Function]}
            >
              <img
                alt="While My Guitar Gently Weeps Live (feat. Gary Clark Jr.) [Live] - Single"
                className="img-fluid rounded"
              />
              <span
                className="text-truncate"
              >
                While My Guitar Gently Weeps Live (feat. Gary Clark Jr.) [Live] - Single
              </span>
            </a>
            <span
              className="text-truncate"
            >
              Will Taylor and Strings Attached
            </span>
          </div>
          <div
            className="sc-AxjAm dYpDUQ"
          >
            <a
              href="/album/1279718975"
              onClick={[Function]}
            >
              <img
                alt="Come Together - Single"
                className="img-fluid rounded"
              />
              <span
                className="text-truncate"
              >
                Come Together - Single
              </span>
            </a>
            <span
              className="text-truncate"
            >
              Gary Clark Jr. & Junkie XL
            </span>
          </div>
          <div
            className="sc-AxjAm dYpDUQ"
          >
            <a
              href="/album/1148774607"
              onClick={[Function]}
            >
              <img
                alt="Take Me Down - Single"
                className="img-fluid rounded"
              />
              <span
                className="text-truncate"
              >
                Take Me Down - Single
              </span>
            </a>
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </div>
          <div
            className="sc-AxjAm dYpDUQ"
          >
            <a
              href="/album/1129970027"
              onClick={[Function]}
            >
              <img
                alt="Jam in the Van - Gary Clark Jr. - EP"
                className="img-fluid rounded"
              />
              <span
                className="text-truncate"
              >
                Jam in the Van - Gary Clark Jr. - EP
              </span>
            </a>
            <span
              className="text-truncate"
            >
              Gary Clark Jr. & Jam in the Van
            </span>
          </div>
          <div
            className="sc-AxjAm dYpDUQ"
          >
            <a
              href="/album/562514719"
              onClick={[Function]}
            >
              <img
                alt="Ain't Messin 'Round - Single"
                className="img-fluid rounded"
              />
              <span
                className="text-truncate"
              >
                Ain't Messin 'Round - Single
              </span>
            </a>
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </div>
        </div>
        <div
          className="sc-AxhCb sc-AxhUy ixXRnl"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
